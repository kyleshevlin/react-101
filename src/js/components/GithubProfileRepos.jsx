import React, { Component } from 'react'
import PropTypes from 'prop-types'

class GithubProfileRepos extends Component {
  state = {
    isVisible: true
  }

  handleClick = () => {
    this.setState({ isVisible: !this.state.isVisible })
  }

  render() {
    const { repos } = this.props

    return (
      <div className="github_profile-repos">
        {repos.length ? (
          <div>
            <button onClick={this.handleClick}>Toggle</button>

            {this.state.isVisible && (
              <ol className="github_profile-repos-list">
                {repos.map(repo => (
                  <li key={repo} className="github_profile-repos-list-item">
                    {repo}
                  </li>
                ))}
              </ol>
            )}
          </div>
        ) : (
          <div className="github_profile-repos-empty">
            <p>No repos were found.</p>
          </div>
        )}
      </div>
    )
  }
}

GithubProfileRepos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default GithubProfileRepos
