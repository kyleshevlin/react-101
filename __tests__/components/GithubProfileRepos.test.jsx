import React from 'react'
import GithubProfileRepos from '../../src/js/components/GithubProfileRepos'

describe('GithubProfileRepos', () => {
  it('should render empty state if no repos', () => {
    const component = shallow(<GithubProfileRepos repos={[]} />)

    expect(component).toMatchSnapshot()
    expect(component.find('.github_profile-repos-empty').length).toEqual(1)
  })

  it('should list repos given to it', () => {
    const repos = ['foo', 'bar']
    const component = shallow(<GithubProfileRepos repos={repos} />)

    expect(component).toMatchSnapshot()
    expect(component.find('.github_profile-repos-list').length).toEqual(1)
    expect(component.find('.github_profile-repos-list-item').length).toEqual(
      repos.length
    )
    expect(
      component
        .find('.github_profile-repos-list-item')
        .first()
        .text()
    ).toEqual(repos[0])
  })

  it('should toggle repos visibility if toggle is clicked', () => {
    const repos = ['foo', 'bar']
    const component = shallow(<GithubProfileRepos repos={repos} />)

    expect(component).toMatchSnapshot()
    expect(component.find('.github_profile-repos-list').length).toEqual(1)

    component.find('button').simulate('click')

    expect(component).toMatchSnapshot()
    expect(component.find('.github_profile-repos-list').length).toEqual(0)

    component.find('button').simulate('click')

    expect(component).toMatchSnapshot()
    expect(component.find('.github_profile-repos-list').length).toEqual(1)
  })
})
