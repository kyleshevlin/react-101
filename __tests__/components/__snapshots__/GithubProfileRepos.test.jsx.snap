// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubProfileRepos should list repos given to it 1`] = `
<div
  className="github_profile-repos"
>
  <div>
    <button
      onClick={[Function]}
    >
      Toggle
    </button>
    <ol
      className="github_profile-repos-list"
    >
      <li
        className="github_profile-repos-list-item"
        key="foo"
      >
        foo
      </li>
      <li
        className="github_profile-repos-list-item"
        key="bar"
      >
        bar
      </li>
    </ol>
  </div>
</div>
`;

exports[`GithubProfileRepos should render empty state if no repos 1`] = `
<div
  className="github_profile-repos"
>
  <div
    className="github_profile-repos-empty"
  >
    <p>
      No repos were found.
    </p>
  </div>
</div>
`;

exports[`GithubProfileRepos should toggle repos visibility if toggle is clicked 1`] = `
<div
  className="github_profile-repos"
>
  <div>
    <button
      onClick={[Function]}
    >
      Toggle
    </button>
    <ol
      className="github_profile-repos-list"
    >
      <li
        className="github_profile-repos-list-item"
        key="foo"
      >
        foo
      </li>
      <li
        className="github_profile-repos-list-item"
        key="bar"
      >
        bar
      </li>
    </ol>
  </div>
</div>
`;

exports[`GithubProfileRepos should toggle repos visibility if toggle is clicked 2`] = `
<div
  className="github_profile-repos"
>
  <div>
    <button
      onClick={[Function]}
    >
      Toggle
    </button>
  </div>
</div>
`;

exports[`GithubProfileRepos should toggle repos visibility if toggle is clicked 3`] = `
<div
  className="github_profile-repos"
>
  <div>
    <button
      onClick={[Function]}
    >
      Toggle
    </button>
    <ol
      className="github_profile-repos-list"
    >
      <li
        className="github_profile-repos-list-item"
        key="foo"
      >
        foo
      </li>
      <li
        className="github_profile-repos-list-item"
        key="bar"
      >
        bar
      </li>
    </ol>
  </div>
</div>
`;
